contrib/platform/src/com/sun/jna/platform/win32/Kernel32Util.java

```diff
@@ -678,4 +680,141 @@ public static final String extractVolumeGUID(String volumeGUIDPath) {
         
         return volumeGUIDPath.substring(VOLUME_GUID_PATH_PREFIX.length(), volumeGUIDPath.length() - VOLUME_GUID_PATH_SUFFIX.length());
     }
+
+	/**
+	 * Gets the specified resource out of the specified executable file
+	 * 
+	 * @param path
+	 *            The path to the executable file
+	 * @param type
+	 *            The type of the resource (either a type name or type ID is
+	 *            allowed)
+	 * @param name
+	 *            The name or ID of the resource
+	 * @return The resource bytes, or null if no such resource exists.
+	 */
+	public static byte[] getResource(String path, String type, String name) {
+		final HMODULE target = Kernel32.INSTANCE.LoadLibraryEx(path, null, Kernel32.LOAD_LIBRARY_AS_DATAFILE);
+
+		Pointer t = null;
+		try {
+			t = new Pointer(Long.parseLong(type));
+		} catch (Exception e) {
+			t = new Memory(Native.WCHAR_SIZE * (type.length() + 1));
+			t.setWideString(0, type);
+		}
+
+		Pointer n = null;
+		try {
+			n = new Pointer(Long.parseLong(name));
+		} catch (Exception e) {
+			n = new Memory(Native.WCHAR_SIZE * (name.length() + 1));
+			n.setWideString(0, name);
+		}
+
+		HRSRC hRsrc = Kernel32.INSTANCE.FindResource(target, n, t);
+		if (hRsrc == null) {
+			return null;
+		}
+
+		HANDLE loaded = Kernel32.INSTANCE.LoadResource(target, hRsrc);
+		int length = Kernel32.INSTANCE.SizeofResource(target, hRsrc);
+		Pointer start = Kernel32.INSTANCE.LockResource(loaded);
+		return start.getByteArray(0, length);
```

Comment:
As per the documentation in https://msdn.microsoft.com/en-us/library/windows/desktop/ms648047(v=vs.85).aspx

>If the loaded resource is available, the return value is a pointer to the first byte of the resource; otherwise, it is NULL.

Your code should check for NULL and handle this somehow - e.g., if NULL is unexpected, then throw some kind of exception - e.g., _IllegalStateException_ - or return NULL - any choice should be **documented** in the _Javadoc_